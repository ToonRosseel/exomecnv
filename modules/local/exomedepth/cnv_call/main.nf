process CNV_CALL {
    tag "$sample $meta2.chr"
    label 'process_low'

    conda "${moduleDir}/environment.yml"
    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
        'https://depot.galaxyproject.org/singularity/r-exomedepth:1.1.16--r43hfb3cda0_3' :
        'biocontainers/r-exomedepth:1.1.16--r43hfb3cda0_3' }"

    publishDir "$params.outdir/exomedepth/cnv_call", mode: 'copy'

    input:
    tuple val(meta), path(exon_target) // meta.id=chrx/autosomal
    tuple val(meta2), val(sample), path(countfile) // meta:id, chr, sam, fam, sample

    output:
    tuple val(meta2), val (sample), path("*.txt"), emit: cnvcall
    path "versions.yml", emit:versions

    script:

    def VERSION = '1.1.16'

    """
    ExomeDepth_cnv_calling.R \\
        $sample \\
        $countfile \\
        $exon_target \\
        $meta2.chr \\
        ${meta2.sam.join(',')} \\
        ${meta2.fam.join(',')} \\

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        ExomeDepth: ${VERSION}
        R: \$(Rscript --version | sed 's/R scripting front-end //g')
    END_VERSIONS
    """

    stub:
    def prefix = task.ext.prefix ?: "${sample}_CNVs_ExomeDepth_${meta2.chr}"
    def VERSION = '1.1.16'
    """
    touch ${prefix}.txt

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        ExomeDepth: ${VERSION}
        R: \$(Rscript --version | sed 's/R scripting front-end //g')
    END_VERSIONS
    """
}
