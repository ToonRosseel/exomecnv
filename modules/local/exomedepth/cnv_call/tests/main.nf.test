nextflow_process {

    name "Test Process CNV_CALL"
    script "../main.nf"
    process "CNV_CALL"

    test("homo_sapiens - bed, merged_counts") {

        when {
            process {
                """
                input[0] = Channel.value([
                    [ id:'exon' ],
                    file(params.roi_auto)
                ])
                input[1] = Channel.of([
                    [ id: 'test', chr:"autosomal", sam:["Sample1", "Sample2", "Sample3", "Sample4", "Sample5"], fam:["Fam1", "Fam2", "Fam3", "Fam4", "Fam5"] ],
                    "Sample1",
                    file(params.merged_counts, checkIfExists:true)
                ])
                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out).match()
        }

    }

    test("homo_sapiens - bed, merged_counts - stub") {

        options "-stub"

        when {
            process {
                """
                input[0] = Channel.value([
                    [ id:'exon' ],
                    file(params.roi_auto)
                ])
                input[1] = Channel.of([
                    [ id: 'test', chr:"autosomal", sam:["Sample1", "Sample2", "Sample3", "Sample4", "Sample5"], fam:["Fam1", "Fam2", "Fam3", "Fam4", "Fam5"] ],
                    "Sample1",
                    file(params.merged_counts, checkIfExists:true)
                ])
                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out).match()
        }

    }

}
