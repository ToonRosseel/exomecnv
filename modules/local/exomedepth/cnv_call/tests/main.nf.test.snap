{
    "Should run without failures": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "chr": "autosomal",
                            "sam": [
                                "Sample1",
                                "Sample2",
                                "Sample3",
                                "Sample4",
                                "Sample5"
                            ],
                            "fam": [
                                "Fam1",
                                "Fam2",
                                "Fam3",
                                "Fam4",
                                "Fam5"
                            ]
                        },
                        "Sample1",
                        "Sample1_CNVs_ExomeDepth_autosomal.txt:md5,cd414909ef75fbabb7a3e33b0ae13ea4"
                    ]
                ],
                "1": [
                    "versions.yml:md5,1c8782fc3a89ec689cfdce0b3f91de85"
                ],
                "cnvcall": [
                    [
                        {
                            "id": "test",
                            "chr": "autosomal",
                            "sam": [
                                "Sample1",
                                "Sample2",
                                "Sample3",
                                "Sample4",
                                "Sample5"
                            ],
                            "fam": [
                                "Fam1",
                                "Fam2",
                                "Fam3",
                                "Fam4",
                                "Fam5"
                            ]
                        },
                        "Sample1",
                        "Sample1_CNVs_ExomeDepth_autosomal.txt:md5,cd414909ef75fbabb7a3e33b0ae13ea4"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,1c8782fc3a89ec689cfdce0b3f91de85"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.1",
            "nextflow": "24.10.0"
        },
        "timestamp": "2024-11-19T16:53:19.291546737"
    }
}